db/compaction.cc

```diff
@@ -296,7 +296,7 @@ bool Compaction::KeyNotExistsBeyondOutputLevel(
     }
     return true;
   } else {
-    return bottommost_level_;
+    return false;
```

Comment:
You could just as well remove else here on line 298, and let the execution fall through to return false. 
Its good either way.